python/lsst/faro/measurement/TractMeasurementTasks.py

```diff
@@ -27,29 +52,30 @@ class WPerpTask(Task):
     _DefaultName = "WPerpTask"
 
     def run(
-        self, metricName, catalogs, photoCalibs=None, astromCalibs=None, dataIds=None
+        self, metricName: str, data: Dict[str, List[CalibratedCatalog]],
     ):
         self.log.info("Measuring %s", metricName)
-        bands = set([f["band"] for f in dataIds])
+        bands = set('gri')
 
-        if ("g" in bands) & ("r" in bands) & ("i" in bands):
-            rgicatAll = makeMatchedPhotom(dataIds, catalogs, photoCalibs)
+        if bands.issubset(set(data.keys())):
```

Comment:
The default iterator for a `dict` is going to be the keys so you only need `set(data)` here I think.

Reply:
The dev guide actually recommends [using dict.keys() when storing keys](https://developer.lsst.io/python/style.html#avoid-dict-keys-when-iterating-over-keys-or-checking-membership), unless I'm reading that wrong.

Reply:
Gosh. I have no idea why since it's not necessary. My only thought is that it's easier for people to see what's happening without having to think about it. cc/ @ktlim

Reply:
I think there's an assumption that converting a view is slightly more performant than converting an iterator.  My timing tests show differences within one standard deviation, so it probably doesn't matter.

Reply:
Also, I agree that the explicitness is desirable.